torchmdnet/neighbors/neighbors_cuda_cell.cuh

```diff
@@ -0,0 +1,409 @@
+/* Raul P. Pelaez 2023. Batched cell list neighbor list implementation for CUDA.
+
+ */
+#ifndef NEIGHBOR_CUDA_CELL_H
+#define NEIGHBOR_CUDA_CELL_H
+#include "common.cuh"
+#include <thrust/extrema.h>
+
+/*
+ * @brief Calculates the cell dimensions for a given box size and cutoff
+ * @param box_size The box size
+ * @param cutoff The cutoff
+ * @return The cell dimensions
+ */
+template <typename scalar_t>
+__host__ __device__ int3 getCellDimensions(scalar3<scalar_t> box_size, scalar_t cutoff) {
+    int3 cell_dim = make_int3(box_size.x / cutoff, box_size.y / cutoff, box_size.z / cutoff);
+    // Minimum 3 cells in each dimension
+    cell_dim.x = thrust::max(cell_dim.x, 3);
+    cell_dim.y = thrust::max(cell_dim.y, 3);
+    cell_dim.z = thrust::max(cell_dim.z, 3);
```

Comment:
AFAIK you are not really allowed to use std::anything in a kernel (see answer [here](https://stackoverflow.com/questions/45488111/how-to-fix-error-calling-a-host-functionstdmaxdouble-is-not-allowed)) because they are __host__ functions. You can sometimes get away with it by abusing that many of these are constexpr, so by passing --expt-relaxed-constexpr nvcc eats them. I try to avoid it because the feature has "experimental" in the name (although after 10 years they might as well stop calling it experimental). Also std::max/min will not call the builtin CUDA overloads and will probably do something like (a>b)?a:b instead. The same is true for other std:: things in kernels.

Thrust aims to solve this by providing alternatives for anything that makes sense, I learned to just default to thrust:: in kernels with time.
Having said this, I can just call the built in min/max overloads directly.